### PR TITLE
Fixing issue146 in unified-builtins2

### PR DESCRIPTION
**Pull request by yilongli**
fixing https://github.com/kframework/k/issues/146 & https://github.com/kframework/k/issues/376

### CODE REVIEW

File: tests/configjavarw.xml [hunk 1]
```diff
@@ -296,11 +297,10 @@
       results="tests/regression/java-rewrite-engine/list/test1/tests"
       extensions="issue"
       >
-    <kompile-option name="-backend" value="java" />
+    <kompile-option name="--backend" value="java" />
```
Comment (dwightguth) [inline tests/configjavarw.xml:300]: @andreistefanescu is this file currently being tested or does he need to add tests he wants to be used to another file somewhere?

File: tests/regression/java-rewrite-engine/list/test1/list-test1.k [hunk 2]
```diff
@@ -1,13 +1,13 @@
 // Copyright (c) 2013-2014 K Team. All Rights Reserved.
```
Comment (dwightguth) [inline tests/regression/java-rewrite-engine/list/test1/list-test1.k:1]: What exactly is being changed in this file?

File: tests/regression/java-rewrite-engine/list/test2/list-test2.k [hunk 3]
```diff
@@ -1,13 +1,13 @@
 // Copyright (c) 2013-2014 K Team. All Rights Reserved.
```
Comment (dwightguth) [inline tests/regression/java-rewrite-engine/list/test2/list-test2.k:1]: What exactly is being changed in this file?

File: tests/regression/java-rewrite-engine/list/test1/list-test1.k (hunk 2 above)
Comment (yilongli) [inline tests/regression/java-rewrite-engine/list/test1/list-test1.k:1]: Probably nothing. I will revert this.

File: tests/regression/java-rewrite-engine/list/test2/list-test2.k (hunk 3 above)
Comment (yilongli) [inline tests/regression/java-rewrite-engine/list/test2/list-test2.k:1]: Same as above.

File: tests/configjavarw.xml (hunk 1 above)
Comment (yilongli) [inline tests/configjavarw.xml:300]: This test was commented out long time ago. I fix it and bring it back.

Comment (dwightguth) [inline tests/configjavarw.xml:300]: Okay, never mind, I see it is actually listed in the build output.

**Comment by andreistefanescu** [comment]
lgtm

**Comment by yilongli** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Pull request merged**